src/GeositeFramework/js/Plugin.js

```diff
@@ -32,7 +32,40 @@ require(['use!Geosite',
                 pluginName = model.get('pluginSrcFolder'),
                 $uiContainer = model.get('$uiContainer'),
                 $legendContainer = model.get('$legendContainer'),
-                logger = new Logger(pluginName);
+                logger = new Logger(pluginName),
+                resizers = {
+                    adjustBy: function(val) {
+                        var currentWidth = parseFloat($uiContainer.css("width"));
+                        var d = $.Deferred();
+                        d.then(function() {
+                            $uiContainer.css({ width: currentWidth + val });
+                        });
+                        return d.resolve();
+                    },
+                    setWidth: function(val) {
+                        var d = $.Deferred(),
+                            setter = function(w) {
+                                return function() {
+                                    $uiContainer.css({ width: "" });
+                                    $uiContainer.removeClass(function() {
+                                        var classes = this.className.split(" ");
+                                        return _.reject(classes, function(c) {
+                                            return /sidebar-width-.*/.test(c);
+                                        });
+                                    });
+                                    $uiContainer.addClass("sidebar-width-" + w);
+                                };
+                            };
+                        if (typeof (val) === "string") {
+                            d.then(setter(val));
+                        } else if (typeof (val) === "number") {
+                            d.then(function() {
+                                $uiContainer.css({ width: val });
+                            });
+                        }
+                        return d.resolve();
```

Comment:
Until we add animated transitions, you could simulate the transition delay by doing something like this:

```
var d = new Deferred();
setTimeout(function() {
    // Set width here
    d.resolve();
}, 1000);
return d.promise;
```